lib/streamlit/hashing.py

```diff
@@ -360,10 +362,13 @@ def _to_bytes(self, obj, context):
                 # it's a callable object that remembers the original function plus
                 # the values you pickled into it. So here we need to special-case it.
                 h = hashlib.new(self.name)
-                self._update(h, obj.args)
-                self._update(h, obj.func)
-                self._update(h, obj.keywords)
+                self._update(h, obj.args, hash_funcs=hash_funcs)
+                self._update(h, obj.func, hash_funcs=hash_funcs)
+                self._update(h, obj.keywords, hash_funcs=hash_funcs)
                 return h.digest()
+            elif type(obj) in hash_funcs:
```

Comment:
I think this branch should be near the top of this `if`. Right below `if isinstance(obj, bytes)`.

The idea is the `hash_funcs` check takes priority over everything (except `bytes` since otherwise we could get in an infinite loop).

Reply:
@tvst Will do.. but why would we want the user to override hashing for a type that we've already implemented hashing for?

Reply:
For example: let's say the user is passing a huge object (among other parameters) into `st.cache`, and this object takes a long time to hash. If the user knows that object is never being mutated, she can make `st.cache` never hash that object by passing `hash_funcs={MyType: (lambda x: 1234)}`